oauth.go

```diff
@@ -270,46 +280,70 @@ func (svc *AlbyOAuthService) AppsNewHandler(c echo.Context) error {
 	svc.db.First(&user, userID)
 
 	return c.Render(http.StatusOK, "apps/new.html", map[string]interface{}{
-		"User": user,
-		"Name": appName,
+		"User":     user,
+		"Name":     appName,
+		"isMobile": checkMobile(c.Request().UserAgent()),
 	})
 }
 
+func checkMobile(userAgent string) bool {
+	//heuristic, will not catch absolutely everything
+	return strings.Contains(userAgent, "Android") || strings.Contains(userAgent, "iPhone") || strings.Contains(userAgent, "iPad")
+}
+
+func (svc *AlbyOAuthService) CreateApp(name string, userId uint) (sk, pk, connectionString string, err error) {
+	user := User{}
+	err = svc.db.Preload("Apps").First(&user, userId).Error
+	if err != nil {
+		return
+	}
+
+	sk = nostr.GeneratePrivateKey()
+	pk, err = nostr.GetPublicKey(sk)
+	if err != nil {
+		return
+	}
+	if name == "" {
+		err = fmt.Errorf("Empty name is not allowed.")
+		return
+	}
+
+	err = svc.db.Model(&user).Association("Apps").Append(&App{Name: name, NostrPubkey: pk})
+	if err != nil {
+		return
+	}
+	connectionString = fmt.Sprintf("nostrwalletconnect://%s?relay=%s&secret=%s", svc.cfg.IdentityPubkey, svc.cfg.Relay, sk)
+	return
+}
+
 func (svc *AlbyOAuthService) AppsCreateHandler(c echo.Context) error {
 	sess, _ := session.Get("alby_nostr_wallet_connect", c)
 	userID := sess.Values["user_id"]
 	if userID == nil {
 		return c.Redirect(302, "/")
 	}
-	user := User{}
-	svc.db.Preload("Apps").First(&user, userID)
-
 	name := c.FormValue("name")
-	var pairingPublicKey string
-	var pairingSecretKey string
-	if c.FormValue("pubkey") == "" {
-		pairingSecretKey = nostr.GeneratePrivateKey()
-		pairingPublicKey, _ = nostr.GetPublicKey(pairingSecretKey)
-	} else {
-		pairingPublicKey = c.FormValue("pubkey")
-	}
-
-	err := svc.db.Model(&user).Association("Apps").Append(&App{Name: name, NostrPubkey: pairingPublicKey})
-	if err == nil {
-		pairingUri := template.URL(fmt.Sprintf("nostrwalletconnect://%s?relay=%s&secret=%s", svc.cfg.IdentityPubkey, svc.cfg.Relay, pairingSecretKey))
-		return c.Render(http.StatusOK, "apps/create.html", map[string]interface{}{
-			"PairingUri":    pairingUri,
-			"PairingSecret": pairingSecretKey,
-			"Pubkey":        pairingPublicKey,
-			"Name":          name,
-		})
-	} else {
+	sk, pk, connectionString, err := svc.CreateApp(name, userID.(uint))
+	if err != nil {
 		svc.Logger.WithFields(logrus.Fields{
-			"pairingPublicKey": pairingPublicKey,
+			"pairingPublicKey": pk,
 			"name":             name,
 		}).Errorf("Failed to save app: %v", err)
 		return c.Redirect(302, "/apps")
 	}
+
+	//if the user is on mobile, redirect them back to their app with the nostrwalletconnect:// deep link
+	if checkMobile(c.Request().UserAgent()) {
```

Comment:
it might even be as easy as: 

```js
const pairingLink = document.getElementById("pairing-link");
pairingLink.click();
```

if no protocol handler is registered nothing happens.